### ISSUE TITLE
Weird yellow background when using Glur on macOS

### ISSUE THREAD

**Issue by mpdifran** [joogps/Glur#8]
Hey all! 

Loving the Glur library, super cool stuff. I've recently integrated it into my App Store screenshot app, and I've noticed some weird behaviour on macOS in particular.

Glur functions as expected on iOS and iPadOS, but when I view the same Glur on macOS, I get a yellow background. Have you seen this before? Any idea what's going on?

Note: The app tint is yellow, so perhaps the yellow is coming from there?

| macOS | iPadOS |
| --- | --- |
|<img width="417" alt="Screenshot 2024-02-17 at 5 06 52 PM" src="https://github.com/joogps/Glur/assets/1365987/253ba3c1-bf2a-4938-86bf-1536f3f554f0">|![IMG_E5B58E6FF948-1](https://github.com/joogps/Glur/assets/1365987/d5ea01f6-f83a-4736-983a-c06ecefd04fe)|

**Comment by joogps** [comment]
hey!

I think I know what this is. when SwiftUI isn't able to rasterize a view (such as through .drawingGroup) it displays a red prohibited icon over a yellow background.

this might be what's happening here, as your background view might not work with .drawingGroup on Mac specifically... is it just a gradient or is there something else?

**Comment by joogps** [comment]
also, this is such a great use of this effect. are you using it to make app previews?

**Comment by mpdifran** [comment]
Yeah that's the plan! I saw Glur, and I was like, this is such a cool effect to have in app screenshots, I have it integrate it haha. 

Let me take a look at what is included in the item I'm blurring. Without revealing too much about the code I'm using to create the device frame, I'm using the following modifiers before applying the Glur effect:

`.opacity`
`Group` to handle if conditions on modifiers
`.shadow`
`.overlay`
`.blendMode`
`.onDrop`
`.simultaneousGesture`
`.onTapGesture`
`.scaleEffect`

I'm drawing the device at full size (somewhere around 1242 x 2688), then using scaleEffect to scale it down in the app, before applying Glur. I was having trouble getting Glur to work before the scale effect, but moving it to after seemed to help. I still have problems for certain large devices even after the scale, like iPads.

Here's my View extension that leverages `Group`:

```
public extension View {

    func `if`<Content: View>(_ condition: Bool, content: (Self) -> Content) -> some View {
        Group {
            if condition {
                content(self)
            } else {
                self
            }
        }
    }
}
```
